src/TraceStream.cc

```diff
@@ -1073,35 +1074,64 @@ void TraceWriter::close() {
   }
 }
 
-static string make_trace_dir(const string& exe_path) {
-  ensure_default_rr_trace_dir();
+static string make_trace_dir(const string& exe_path, const string& output_trace_dir) {
+  // save trace dir in given output trace dir with option -o
+  if (!output_trace_dir.empty()) {
+    DIR* directory = opendir(output_trace_dir.c_str());
+    if (directory) {
+	  // directory exists already
+      FATAL() << "Directory `" << output_trace_dir << "' exists already. Could not make it to trace directory!";
+    }
+	else if (ENOENT == errno) {
+      // directory does not exist, create it
+      int ret = mkdir(output_trace_dir.c_str(), S_IRWXU | S_IRWXG);
```

Comment:
Why not just call `mkdir` and print the above message if it returns `EEXIST`? That also eliminates a race condition where something creates the directory between rr trying to open it and calling `mkdir`.